core/sdk-core/src/main/java/software/amazon/awssdk/core/client/handler/BaseClientHandler.java

```diff
@@ -185,4 +203,36 @@ protected boolean isCalculateCrc32FromCompressedData() {
         HttpResponseHandler<OutputT> interceptorCallingResponseHandler = interceptorCalling(delegate, executionContext);
         return new AttachHttpMetadataResponseHandler<>(interceptorCallingResponseHandler);
     }
+
+    /**
+     * Sets the metrics registry instance in execution attributes.
+     * Register some constant metrics regarding the client and the API
+     */
+    protected void initializeMetrics(ExecutionContext executionContext) {
+        MetricRegistry metricRegistry = null;
```

Comment:
Initializing per request? Anything we can save here?

Reply:
Each request will have its own MetricRegistry. This helps in storing retry info in the same top-level MetricRegistry instance